.github/actions/e2e_mini/action.yml

```diff
@@ -38,9 +41,26 @@ runs:
 
     - name: MiniConstellation E2E
       shell: bash
+      id: e2e-test
       env:
         ARM_CLIENT_ID: ${{ inputs.azureClientID }}
         ARM_SUBSCRIPTION_ID: ${{ inputs.azureSubscriptionID }}
         ARM_TENANT_ID: ${{ inputs.azureTenantID }}
       run: |
         bazel run --test_timeout=14400 //e2e/miniconstellation:push_remote_test
+
+    - name: Log in to azure
+      # only log in if e2e test failed or if the run was cancelled
+      if: (failure() && steps.e2e-test.conclusion == 'failure') || cancelled()
+      uses: ./.github/actions/login_azure
+      with:
+        azure_credentials: ${{ inputs.azureIAMCredentials }}
+
+    - name: Clean up after failure
```

Comment:
Why do we clean up via the az CLI and not via terraform?

Reply:
I originally planned to clean up via terraform but encountered some problems on the way, so I settled for the azure CLI as an alternative solution. Also, it may be that terraform fails itself (for some reason), so I believe it would be good to have an alternative approach that is independent of the terraform state as well.

Reply:
I would keep it with just deleting the resource group:
The test is heavily dependent on running inside an Azure VM currently, so I don't think its worth the effort to generalize that part.
Additionally, to effectively clean up the infrastructure using Terraform, the whole test setup needs to be rewritten since it currently happens as part of the testing script, not as its own step in the workflow.